webapp/channels/src/components/integrations/abstract_oauth_app.tsx

```diff
@@ -1,74 +1,94 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
+import type {ChangeEvent, FormEvent} from 'react';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
+import type {MessageDescriptor} from 'react-intl';
 import {Link} from 'react-router-dom';
 
+import type {OAuthApp} from '@mattermost/types/integrations';
+import type {Team} from '@mattermost/types/teams';
+
 import {Permissions} from 'mattermost-redux/constants';
 
 import BackstageHeader from 'components/backstage/components/backstage_header';
 import FormError from 'components/form_error';
 import SystemPermissionGate from 'components/permissions_gates/system_permission_gate';
 import SpinnerButton from 'components/spinner_button';
 
-import {localizeMessage} from 'utils/utils';
-
-export default class AbstractOAuthApp extends React.PureComponent {
-    static propTypes = {
-
-        /**
-        * The current team
-        */
-        team: PropTypes.object.isRequired,
-
-        /**
-        * The header text to render, has id and defaultMessage
-        */
-        header: PropTypes.object.isRequired,
-
-        /**
-        * The footer text to render, has id and defaultMessage
-        */
-        footer: PropTypes.object.isRequired,
-
-        /**
-        * The spinner loading text to render, has id and defaultMessage
-        */
-        loading: PropTypes.object.isRequired,
-
-        /**
-         * Any extra component/node to render
-         */
-        renderExtra: PropTypes.node.isRequired,
-
-        /**
-        * The server error text after a failed action
-        */
-        serverError: PropTypes.string.isRequired,
-
-        /**
-        * The OAuthApp used to set the initial state
-        */
-        initialApp: PropTypes.object,
-
-        /**
-        * The async function to run when the action button is pressed
-        */
-        action: PropTypes.func.isRequired,
-    };
+import * as Utils from 'utils/utils';
+
+type Props = {
+
+    /**
+   * The current team
+   */
+    team: Team;
+
+    /**
+   * The header text to render, has id and defaultMessage
+   */
+    header: MessageDescriptor | string;
+
+    /**
+   * The footer text to render, has id and defaultMessage
+   */
+    footer: MessageDescriptor | string;
+
+    /**
+   * The spinner loading text to render, has id and defaultMessage
+   */
+    loading: MessageDescriptor | string;
```

Comment:
Do we need the `string` type for `header`, `footer` and `loading`? Because when I check the usage in `EditOAuthApp` and `AddOAuthApp`, always passed as object.